STX.Serialization.Providers.SystemTextJson/Services/Foundations/Serializations/SerializationService.Validations.cs

```diff
@@ -15,5 +17,67 @@ private static void ValidateInputIsNotNull<TInput>(TInput @object)
                 throw new NullSerializationException(message: $"Input is null.");
             }
         }
+
+        private static void ValidateInput<TInput>(TInput json)
+        {
+            switch (typeof(TInput))
+            {
+                case Type _ when typeof(TInput) == typeof(string):
```

Comment:
Instead Type _ you could do,

case Type inputType when inputType == typeof(string)